prow/config/config.go

```diff
@@ -202,11 +203,9 @@ type Plank struct {
 	// PodPendingTimeout is after how long the controller will perform a garbage
 	// collection on pending pods. Defaults to one day.
 	PodPendingTimeout time.Duration `json:"-"`
-	// PodRunningTimeoutString compiles into PodRunningTimeout at load time.
-	PodRunningTimeoutString string `json:"pod_running_timeout,omitempty"`
 	// PodRunningTimeout is after how long the controller will abort a prowjob pod
 	// stuck in running state. Defaults to two days.
-	PodRunningTimeout time.Duration `json:"-"`
+	PodRunningTimeout *metav1.Duration `json:"pod_running_timeout,omitempty"`
```

Comment:
I don't think we need to distinguish `0` from "unset" for any of these duration fields so we should avoid using pointer fields so that we don't have to update all the existing uses of these fields with `nil` checks.

Reply:
Don't we need pointers to allow `omitempty` thought?

Reply:
(and we need `omitempty` for everyone that's serializing config)

Reply:
https://github.com/kubernetes/test-infra/pull/12414

Reply:
emm, we are ensuring all these fields will be non-nil after we check the default values though. 

(Although Glanced through, looks like 0 is not a valid value for any of our dual duration fields, I'm happy with either way)

Reply:
emm, yeah, that